client/components/domains/registrant-extra-info/uk-form.jsx

```diff
@@ -0,0 +1,198 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { camelCase, difference, includes, isEmpty, keys, map, pick } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getContactDetailsCache } from 'state/selectors';
+import { getCurrentUserLocale } from 'state/current-user/selectors';
+import { updateContactDetailsCache } from 'state/domains/management/actions';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSelect from 'components/forms/form-select';
+import FormTextInput from 'components/forms/form-text-input';
+
+const defaultValues = {
+	registrantType: 'IND',
+};
+
+class RegistrantExtraInfoUkForm extends React.PureComponent {
+	static propTypes = {
+		contactDetails: PropTypes.object.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+		const { translate } = props;
+		const registrantTypes = {
+			IND: translate( 'Individual' ),
+			FIND: translate( 'Foreign Individual' ),
+			STRA: translate( 'UK Sole Trader', {
+				comment: 'Refers to UK legal concept of self-employment/sole proprietorship',
+			} ),
+			PTNR: translate( 'UK Partnership' ),
+			LLP: translate( 'UK Limited Liability Partnership' ),
+			CRC: translate( 'UK Corporation by Royal Charter' ),
+			FCORP: translate( 'Non-UK Corporation' ),
+			IP: translate( 'UK Industrial/Provident Registered Company' ),
+			LTD: translate( 'UK Limited Company' ),
+			PLC: translate( 'UK Public Limited Company' ),
+			SCH: translate( 'UK School' ),
+			GOV: translate( 'UK Government Body' ),
+			RCHAR: translate( 'UK Registered Charity' ),
+			STAT: translate( 'UK Statutory Body' ),
+			OTHER: translate( 'UK Entity that does not fit another category' ),
+			FOTHER: translate( 'Non-UK Entity that does not fit another category' ),
+		};
+		const registrantTypeOptions = map( registrantTypes, ( text, optionValue ) => (
+			<option value={ optionValue } key={ optionValue }>
+				{ text }
+			</option>
+		) );
+
+		this.state = {
+			registrantTypes,
+			registrantTypeOptions,
+		};
+	}
+
+	componentWillMount() {
+		// Add defaults to redux state to make accepting default values work.
+		const neededRequiredDetails = difference(
```

Comment:
I'm pretty sure this is copied from code I wrote, but I don't love it. I wonder if there's a cleaner way to solve this problem (to whit: hitting submit without updating any values on the form).

OTOH, if it aint broke...